### PR TITLE
[release-3.6] Bug 1534588 - Size of the emitted data exceeds buffer_chunk_limit

### PR DESCRIPTION
**Pull request by nhosoi**
If secure-forward outputs are additionally configured, the output
count is also to be added to NUM_OUTPUTS, which is a divider to
calculate the buffer size per output.

Adding tests to cover the cases - having 1 then 2 intermediate
forwarding fluentds.

Manually cherry-picked:
a24301bb9ae6bdfe3bca934a24c79ed68b1e8164 (origin/pr/926, nhosoi/bz1534588, bz1534588)

### CODE REVIEW

File: test/fluentd-forward.sh [hunk 1]
```diff
@@ -10,74 +10,175 @@ FLUENTD_WAIT_TIME=${FLUENTD_WAIT_TIME:-$(( 2 * minute ))}
 
 os::test::junit::declare_suite_start "test/fluentd-forward"
 
+extra_artifacts=$ARTIFACT_DIR/fluentd-forward-artifacts.txt
```
Comment (richm) [inline test/fluentd-forward.sh]: use `artifact_out` and `artifact_log`

File: test/fluentd-forward.sh [hunk 2]
```diff
@@ -10,74 +10,175 @@ FLUENTD_WAIT_TIME=${FLUENTD_WAIT_TIME:-$(( 2 * minute ))}
...
     # undeploy fluentd
     os::log::debug "$( oc label node --all logging-infra-fluentd- )"
     os::cmd::try_until_text "oc get daemonset logging-fluentd -o jsonpath='{ .status.numberReady }'" "0" $FLUENTD_WAIT_TIME
 
+    FLUENTD_FORWARD=()
+    id=0
+    while [ $id -lt $cnt ]; do
+      POD=$( oc get pods -l component=forward-fluentd${id} -o name )
+      FLUENTD_FORWARD[$id]=$( oc get $POD --template='{{.status.podIP}}' )
+      echo "update_current_fluentd .status.podIP ${FLUENTD_FORWARD[$id]}" >> $extra_artifacts
```
Comment (richm) [inline test/fluentd-forward.sh]: `artifact_log update_current_fluentd .status.podIP ${FLUENTD_FORWARD[$id]}`

File: test/fluentd-forward.sh (hunk 1 above)
Comment (nhosoi) [reply test/fluentd-forward.sh]: This is a backport (source: https://github.com/openshift/origin-aggregated-logging/pull/926)
Should I fix the newer versions, too?

Comment (richm) [reply test/fluentd-forward.sh]: arg - sorry, I missed the review of the earlier PRs - then no, don't fix it here

instead, submit another PR (later) to fix the test in master and cherry pick from there

Comment (nhosoi) [reply test/fluentd-forward.sh]: Yep.  I've just submitted this PR: https://github.com/openshift/origin-aggregated-logging/pull/946

**Pull request merged**